### PR TITLE
Info only mode

### PR DESCRIPTION
**Pull request by JonathanAndrews**
- Add a mode that does not redirect the user if they aren't recognised, but does log the results of the Recognition analysis.

### CODE REVIEW

**Review by em-cd — COMMENTED** [review]
Couple of comments inline!

File: app/controllers/devise_recognisable/sessions_controller.rb [hunk 1]
```diff
@@ -16,18 +16,22 @@ def perform_recognition_check
...
-      resource_class.send_new_ip_email(resource_params)
+        # Send an email to the user with a sign in link containing a unique
+        # token that is valid for 5 minutes.
+        resource_class.send_new_ip_email(resource_params)
+      end
 
-      # Debug mode only
-      if Devise.debug_mode && Rails.env.production?
+      # debug or info_only mode
+      if (Devise.debug_mode || Devise.info_only) && Rails.env.production?
```
[the remark this answers is not in the document]
Comment (JonathanAndrews) [reply app/controllers/devise_recognisable/sessions_controller.rb:34]: I'm not sure what you mean here Lizzie.
Who's tests? The developer using the gem? Why would they be testing the gem?

[the remark this answers is not in the document]
Comment (em-cd) [reply app/controllers/devise_recognisable/sessions_controller.rb:34]: Maybe I haven't understood what's going on here! I understood that if you set `info_only` when using the gem in your app, in _production_ environments your users won't be asked to sign in via an email link. But in your test and development environments, it's possible they will be asked to. Have I got it mixed up?

[the remark this answers is not in the document]
Comment (JonathanAndrews) [reply app/controllers/devise_recognisable/sessions_controller.rb:34]: Ah ok.
On line 22, we check `Devise.info_only`. This skips the "sign in via an email link" journey for all users in all environments, `development`, `test` or `production`.
On line 34, we check that we are running in `production` before sending the debug information to Rollbar.
So there are two checks for the two different pieces of behaviour: Emails stuff and Logging.
Does that make sense?

[the remark this answers is not in the document]
Comment (em-cd) [reply app/controllers/devise_recognisable/sessions_controller.rb:34]: Ahhh you're so right! 👍 👍 👍

**Review by em-cd — APPROVED** [review]

**Pull request merged**